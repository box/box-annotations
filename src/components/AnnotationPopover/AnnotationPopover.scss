@import '../../variables';

.ba {
    .ba-popover {
        cursor: auto; // Overrides cursor: none from annotation mode
        max-width: 282px;
        position: absolute;
        z-index: 3;
    }

    .ba-popover-caret {
        left: 50%;
        position: absolute;
        z-index: 3;

        // CSS arrow for caret above container
        &::after,
        &::before {
            border: solid transparent;
            content: ' ';
            height: 0;
            left: 50%;
            pointer-events: none;
            position: absolute;
            top: 0;
            transform: rotate(-225deg);
            width: 0;
        }

        &::after {
            border-color: transparent transparent $white $white;
            border-width: 6px;
            margin: -4px 0 0 -6px;
        }

        &::before {
            border-color: transparent transparent $sf-fog $sf-fog;
            border-width: 6px;
            margin: -5px 0 0 -6px;
        }
    }

    .overlay {
        border: 1px solid $seesee;
        border-radius: 4px;
        display: block;
        overflow-x: hidden;
        overflow-y: auto;
        padding: 15px;
        white-space: normal;
    }

    .ba-inline .overlay {
        display: inline-flex;
    }

    .bcs-comment-input-form-container {
        margin-right: 0;

        form {
            margin: 0;
        }
    }

    .bcs-comment-input {
        margin-top: 0;

        ~ .bcs-comment-input-controls .btn {
            margin-bottom: 0;
        }
    }

    .ba-buffer {
        display: none;
    }

    .ba-popover-flipped {
        path,
        rect {
            display: none;
        }

        .ba-popover-caret {
            bottom: 0;
            transform: rotate(180deg);
        }
    }

    // MOBILE & TABLET CSS
    @media #{$mobile}, #{$tablet} {
        // SAVED ANNOTATION POPOVER CSS
        .ba-popover {
            display: flex;
            flex-direction: column;
            height: 100%;
            justify-content: flex-start;
            max-width: unset;
            top: 0;
            width: 100%;
        }

        .ba-mobile-header {
            background: white;
            border-bottom: 1px solid #ccc;
            display: flex;
            justify-content: flex-end;
            padding-right: 5px;
            width: 100%;
        }

        .ba-popover-overlay {
            background: white;
            display: flex;
            flex-direction: column;
            height: 100%;
        }

        .ba-popover .overlay {
            border: none;
            box-shadow: unset;
            display: flex;
            flex-direction: column;
            height: 100%;
            overflow: unset;
            padding: 0;
        }

        // INLINE ANNOTATION POPOVER CSS
        .ba-popover.ba-inline {
            height: auto;
        }

        .ba-inline .overlay {
            background: transparent;
            border-bottom: 1px solid $seesee;
            box-shadow: 0 2px 6px 0 rgba(0, 0, 0, .1);
            height: auto;
            margin: 0;
            text-align: center;
        }

        .ba-inline.ba-create-popover .overlay {
            border: none;
            box-shadow: none;
            display: flex;
            flex-direction: column;
        }

        .ba-inline .ba-mobile-header,
        .ba-create-popover .ba-mobile-header {
            display: none;
        }

        .ba-create-popover .ba-popover-overlay,
        .ba-inline .ba-popover-overlay {
            background: transparent;
            height: auto;
        }

        // UNSAVED/CREATE ANNOTATION POPOVER CSS
        .ba-create-popover {
            bottom: 0;
            height: auto;
            top: auto;
        }

        .ba-create-popover .overlay {
            background: transparent;
            border-radius: 0;
            height: auto;
            justify-content: flex-end;
            padding: 0;
        }
    }

    // TABLET-SPECIFIC CSS
    @media #{$tablet} {
        .ba-popover {
            border-left: 1px solid $seesee;
            // Max width matches the max-width of the Preview Sidebar
            max-width: 341px;
            right: 0;
        }

        .ba-create-popover .ba-mobile-header {
            display: flex;
        }

        .ba-popover.ba-inline {
            border: none;
            width: 100%;
        }

        .ba-inline .ba-mobile-header {
            display: none;
        }

        .ba-create-popover,
        .ba-create-popover .overlay {
            background: white;
            height: 100%;
        }

        .ba-inline.ba-create-popover,
        .ba-inline.ba-create-popover .overlay {
            background: transparent;
            height: auto;
        }

        .ba-create-popover .overlay {
            padding: 15px;
        }
    }
}
