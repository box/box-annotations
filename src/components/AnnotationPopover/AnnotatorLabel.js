// @flow
import React from 'react';
import { injectIntl } from 'react-intl';
import get from 'lodash/get';

import CommentText from '../../../third-party/components/Comment/CommentText';

import { isHighlightAnnotation, isDrawingAnnotation } from '../../util';
import messages from './messages';

import './AnnotatorLabel.scss';

type Props = {
    id?: string,
    type: AnnotationType,
    createdBy: ?User,
    isPending: boolean,
    intl: any
};

class AnnotatorLabel extends React.PureComponent<Props> {
    /**
     * Return a string describes the action completed by the annotator
     * @return {string} Localized annotator label message
     */
    getAnnotatorLabelMessage(): string {
        const { type, createdBy, intl } = this.props;
        const anonymousUserName = intl.formatMessage(messages.anonymousUserName);
        const name = get(createdBy, 'name', anonymousUserName);

        let labelMessage = messages.whoAnnotated;
        if (isHighlightAnnotation(type)) {
            labelMessage = messages.whoHighlighted;
        } else if (isDrawingAnnotation(type)) {
            labelMessage = messages.whoDrew;
        }

        return intl.formatMessage(labelMessage, { name });
    }

    render() {
        const { id, isPending } = this.props;
        return (
            !isPending && (
                <span className='ba-annotator-label'>
                    <CommentText id={id} tagged_message={this.getAnnotatorLabelMessage()} translationEnabled={false} />
                </span>
            )
        );
    }
}

export default injectIntl(AnnotatorLabel);
