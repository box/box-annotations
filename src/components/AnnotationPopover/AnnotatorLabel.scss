@import "../../variables";

.ba {
    .ba-annotator-label {
        margin: 10px 0 10px 10px;
        padding-right: 10px;
        border-right: 1px solid $see-see;
    }

    // MOBILE & TABLET CSS
    @media #{$mobile}, #{$tablet} {
        .ba-annotator-label {
            position: fixed;
            top: 0;
            left: 0;
            display: flex;
            justify-content: center;
            width: 100%;
            padding: 20px;
            background: white;
            border-bottom: 1px solid #ccc;
        }
    }
}
