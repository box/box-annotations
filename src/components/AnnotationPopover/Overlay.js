// @flow
import * as React from 'react';

import OverlayComponent from 'box-react-ui/lib/components/flyout/Overlay';

const CLASS_POPOVER_OVERLAY = 'ba-popover-overlay';

type Props = {
    shouldDefaultFocus: boolean,
    children: React.Node
};

const Overlay = ({ shouldDefaultFocus, children }: Props) => {
    if (shouldDefaultFocus) {
        return (
            <OverlayComponent className={CLASS_POPOVER_OVERLAY} shouldOutlineFocus={false}>
                {children}
            </OverlayComponent>
        );
    }

    return <div className={CLASS_POPOVER_OVERLAY}>{children}</div>;
};

export default Overlay;
