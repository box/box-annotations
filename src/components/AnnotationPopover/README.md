### Description
Component to represent a popover for a specified annotation thread

### Examples
```jsx
const AnnotationPopoverExample = require('examples').AnnotationPopoverExample;

<AnnotationPopoverExample />
```
