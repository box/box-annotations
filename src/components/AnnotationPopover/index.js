// @flow
export { default } from './AnnotationPopover';
