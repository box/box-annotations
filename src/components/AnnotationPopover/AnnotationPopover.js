// @flow
import React from 'react';
import classNames from 'classnames';
import noop from 'lodash/noop';
import PlainButton from 'box-react-ui/lib/components/plain-button';
import IconClose from 'box-react-ui/lib/icons/general/IconClose';

import Internationalize from '../Internationalize';
import Overlay from './Overlay';
import CommentList from '../CommentList';
import { TYPES, CLASS_ANNOTATION_POPOVER } from '../../constants';

import './AnnotationPopover.scss';
import ActionControls from '../ActionControls';
import AnnotatorLabel from './AnnotatorLabel';

type Props = {
    isMobile: boolean,
    canComment: boolean,
    position: Function,
    onDelete: Function,
    onCancel: Function,
    onCreate: Function,
    onCommentClick: Function,
    isPending: boolean,
    language?: string,
    messages?: StringMap,
    headerHeight?: number
} & Annotation;

class AnnotationPopover extends React.PureComponent<Props> {
    static defaultProps = {
        isMobile: false,
        isPending: false,
        canAnnotate: false,
        canComment: false,
        canDelete: false,
        onCommentClick: noop,
        onDelete: noop,
        onCancel: noop,
        onCreate: noop,
        comments: []
    };

    componentDidMount() {
        const { position } = this.props;
        position();
    }

    componentDidUpdate() {
        const { position } = this.props;
        position();
    }

    render() {
        const {
            id,
            type,
            createdAt,
            createdBy,
            comments,
            isMobile,
            canComment,
            canAnnotate,
            isPending,
            canDelete,
            onDelete,
            onCancel,
            onCreate,
            onCommentClick,
            language,
            messages: intlMessages,
            headerHeight
        } = this.props;
        const hasComments = comments.length > 0;
        const isInline = !hasComments && (type === TYPES.highlight || type === TYPES.draw);

        return (
            <Internationalize language={language} messages={intlMessages}>
                <div
                    className={classNames(CLASS_ANNOTATION_POPOVER, {
                        'ba-inline': isInline,
                        'ba-animate-popover': isMobile,
                        'ba-create-popover': isPending
                    })}
                >
                    {isMobile ? (
                        <span className='ba-mobile-header' style={{ height: headerHeight }}>
                            <PlainButton className='ba-mobile-close-btn' onClick={onCancel}>
                                <IconClose height={24} width={24} />
                            </PlainButton>
                        </span>
                    ) : (
                        <span className='ba-popover-caret' />
                    )}
                    <Overlay shouldDefaultFocus={!isMobile}>
                        {hasComments ? (
                            <CommentList comments={comments} onDelete={onDelete} />
                        ) : (
                            <AnnotatorLabel id={id} type={type} createdBy={createdBy} isPending={isPending} />
                        )}
                        {canAnnotate && (
                            <ActionControls
                                id={id}
                                type={type}
                                hasComments={hasComments}
                                isPending={isPending}
                                canComment={canComment}
                                canDelete={canDelete}
                                createdBy={createdBy}
                                createdAt={createdAt}
                                onCreate={onCreate}
                                onCancel={onCancel}
                                onDelete={onDelete}
                                onCommentClick={onCommentClick}
                            />
                        )}
                    </Overlay>
                </div>
            </Internationalize>
        );
    }
}

export default AnnotationPopover;
