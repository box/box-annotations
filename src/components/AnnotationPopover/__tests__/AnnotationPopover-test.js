import * as React from 'react';
import { shallow } from 'enzyme';

import AnnotationPopover from '../AnnotationPopover';
import { TYPES } from '../../../constants';

const fnMock = jest.fn();

const TIME_STRING_SEPT_27_2017 = '2017-09-27T10:40:41-07:00';

const USER = {
    type: 'user',
    id: '789ghi',
    name: 'Mia Thermopolis',
    email: 'princess@genovia.gov'
};

const comments = [
    {
        id: '123',
        createdAt: TIME_STRING_SEPT_27_2017,
        createdBy: USER,
        message: 'test',
        permissions: {}
    },
    {
        id: '456',
        createdAt: TIME_STRING_SEPT_27_2017,
        createdBy: USER,
        message: 'test',
        permissions: {}
    }
];

describe('components/AnnotationPopover', () => {
    const render = (props = {}) =>
        shallow(
            <AnnotationPopover
                canAnnotate={false}
                canDelete={false}
                onCreate={fnMock}
                onCancel={fnMock}
                onDelete={fnMock}
                position={fnMock}
                {...props}
            />
        );

    test('should correctly render a view-only popover with comments', () => {
        const wrapper = render({
            comments
        });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.ba-inline').length).toEqual(0);
    });

    test('should render a view-only annotation with a annotator label and no comments', () => {
        const wrapper = render({
            comments: [],
            type: TYPES.highlight
        });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.ba-inline').length).toEqual(1);
    });

    test('should correctly render an annotation with a annotator label and no comments', () => {
        const wrapper = render({
            canAnnotate: true,
            comments: [],
            type: TYPES.highlight
        });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.ba-inline').length).toEqual(1);
    });

    test('should correctly render a BRUI Overlay if not on mobile', () => {
        const wrapper = render({
            canAnnotate: true,
            comments,
            isMobile: false
        });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('Overlay').prop('shouldDefaultFocus')).toBeTruthy();
    });

    test('should correctly render a div without a Focus Trap if on mobile', () => {
        const wrapper = render({
            canAnnotate: true,
            comments,
            isMobile: true
        });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('Overlay').prop('shouldDefaultFocus')).toBeFalsy();
    });

    test('should correctly render a popover with comments and reply textarea', () => {
        const wrapper = render({
            canAnnotate: true,
            comments
        });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.ba-inline').length).toEqual(0);
    });

    test('should correctly render a pending annotation', () => {
        const wrapper = render({
            isPending: true,
            canAnnotate: true
        });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.ba-inline').length).toEqual(0);
    });
});
