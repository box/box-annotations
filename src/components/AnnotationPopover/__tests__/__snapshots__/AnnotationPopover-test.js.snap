// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AnnotationPopover should correctly render a BRUI Overlay if not on mobile 1`] = `
<Internationalize>
  <div
    className="ba-popover"
  >
    <span
      className="ba-popover-caret"
    />
    <Overlay
      shouldDefaultFocus={true}
    >
      <CommentList
        comments={
          Array [
            Object {
              "createdAt": "2017-09-27T10:40:41-07:00",
              "createdBy": Object {
                "email": "princess@genovia.gov",
                "id": "789ghi",
                "name": "Mia Thermopolis",
                "type": "user",
              },
              "id": "123",
              "message": "test",
              "permissions": Object {},
            },
            Object {
              "createdAt": "2017-09-27T10:40:41-07:00",
              "createdBy": Object {
                "email": "princess@genovia.gov",
                "id": "789ghi",
                "name": "Mia Thermopolis",
                "type": "user",
              },
              "id": "456",
              "message": "test",
              "permissions": Object {},
            },
          ]
        }
        onDelete={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
      />
      <ActionControls
        canComment={false}
        canDelete={false}
        hasComments={true}
        isPending={false}
        onCancel={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        onCommentClick={[Function]}
        onCreate={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        onDelete={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
      />
    </Overlay>
  </div>
</Internationalize>
`;

exports[`components/AnnotationPopover should correctly render a div without a Focus Trap if on mobile 1`] = `
<Internationalize>
  <div
    className="ba-popover ba-animate-popover"
  >
    <span
      className="ba-mobile-header"
      style={
        Object {
          "height": undefined,
        }
      }
    >
      <PlainButton
        className="ba-mobile-close-btn"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
      >
        <IconClose
          height={24}
          width={24}
        />
      </PlainButton>
    </span>
    <Overlay
      shouldDefaultFocus={false}
    >
      <CommentList
        comments={
          Array [
            Object {
              "createdAt": "2017-09-27T10:40:41-07:00",
              "createdBy": Object {
                "email": "princess@genovia.gov",
                "id": "789ghi",
                "name": "Mia Thermopolis",
                "type": "user",
              },
              "id": "123",
              "message": "test",
              "permissions": Object {},
            },
            Object {
              "createdAt": "2017-09-27T10:40:41-07:00",
              "createdBy": Object {
                "email": "princess@genovia.gov",
                "id": "789ghi",
                "name": "Mia Thermopolis",
                "type": "user",
              },
              "id": "456",
              "message": "test",
              "permissions": Object {},
            },
          ]
        }
        onDelete={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
      />
      <ActionControls
        canComment={false}
        canDelete={false}
        hasComments={true}
        isPending={false}
        onCancel={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        onCommentClick={[Function]}
        onCreate={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        onDelete={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
      />
    </Overlay>
  </div>
</Internationalize>
`;

exports[`components/AnnotationPopover should correctly render a pending annotation 1`] = `
<Internationalize>
  <div
    className="ba-popover ba-create-popover"
  >
    <span
      className="ba-popover-caret"
    />
    <Overlay
      shouldDefaultFocus={true}
    >
      <AnnotatorLabel
        isPending={true}
      />
      <ActionControls
        canComment={false}
        canDelete={false}
        hasComments={false}
        isPending={true}
        onCancel={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        onCommentClick={[Function]}
        onCreate={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        onDelete={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
      />
    </Overlay>
  </div>
</Internationalize>
`;

exports[`components/AnnotationPopover should correctly render a popover with comments and reply textarea 1`] = `
<Internationalize>
  <div
    className="ba-popover"
  >
    <span
      className="ba-popover-caret"
    />
    <Overlay
      shouldDefaultFocus={true}
    >
      <CommentList
        comments={
          Array [
            Object {
              "createdAt": "2017-09-27T10:40:41-07:00",
              "createdBy": Object {
                "email": "princess@genovia.gov",
                "id": "789ghi",
                "name": "Mia Thermopolis",
                "type": "user",
              },
              "id": "123",
              "message": "test",
              "permissions": Object {},
            },
            Object {
              "createdAt": "2017-09-27T10:40:41-07:00",
              "createdBy": Object {
                "email": "princess@genovia.gov",
                "id": "789ghi",
                "name": "Mia Thermopolis",
                "type": "user",
              },
              "id": "456",
              "message": "test",
              "permissions": Object {},
            },
          ]
        }
        onDelete={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
      />
      <ActionControls
        canComment={false}
        canDelete={false}
        hasComments={true}
        isPending={false}
        onCancel={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        onCommentClick={[Function]}
        onCreate={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        onDelete={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
      />
    </Overlay>
  </div>
</Internationalize>
`;

exports[`components/AnnotationPopover should correctly render a view-only popover with comments 1`] = `
<Internationalize>
  <div
    className="ba-popover"
  >
    <span
      className="ba-popover-caret"
    />
    <Overlay
      shouldDefaultFocus={true}
    >
      <CommentList
        comments={
          Array [
            Object {
              "createdAt": "2017-09-27T10:40:41-07:00",
              "createdBy": Object {
                "email": "princess@genovia.gov",
                "id": "789ghi",
                "name": "Mia Thermopolis",
                "type": "user",
              },
              "id": "123",
              "message": "test",
              "permissions": Object {},
            },
            Object {
              "createdAt": "2017-09-27T10:40:41-07:00",
              "createdBy": Object {
                "email": "princess@genovia.gov",
                "id": "789ghi",
                "name": "Mia Thermopolis",
                "type": "user",
              },
              "id": "456",
              "message": "test",
              "permissions": Object {},
            },
          ]
        }
        onDelete={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
      />
    </Overlay>
  </div>
</Internationalize>
`;

exports[`components/AnnotationPopover should correctly render an annotation with a annotator label and no comments 1`] = `
<Internationalize>
  <div
    className="ba-popover ba-inline"
  >
    <span
      className="ba-popover-caret"
    />
    <Overlay
      shouldDefaultFocus={true}
    >
      <AnnotatorLabel
        isPending={false}
        type="highlight"
      />
      <ActionControls
        canComment={false}
        canDelete={false}
        hasComments={false}
        isPending={false}
        onCancel={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        onCommentClick={[Function]}
        onCreate={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        onDelete={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
        type="highlight"
      />
    </Overlay>
  </div>
</Internationalize>
`;

exports[`components/AnnotationPopover should render a view-only annotation with a annotator label and no comments 1`] = `
<Internationalize>
  <div
    className="ba-popover ba-inline"
  >
    <span
      className="ba-popover-caret"
    />
    <Overlay
      shouldDefaultFocus={true}
    >
      <AnnotatorLabel
        isPending={false}
        type="highlight"
      />
    </Overlay>
  </div>
</Internationalize>
`;
